// vue-ts-types/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isNonPositive = exports.isNonNegative = exports.isPositive = exports.isNegative = exports.instanceOfProp = exports.oneOfTypesProp = exports.oneOfObjectKeysProp = exports.oneOfProp = exports.functionProp = exports.objectProp = exports.arrayProp = exports.anyProp = exports.vueComponentProp = exports.symbolProp = exports.integerProp = exports.numberProp = exports.booleanProp = exports.stringProp = void 0;
var string_1 = require("./prop-types/string");
Object.defineProperty(exports, "stringProp", { enumerable: true, get: function () { return string_1.stringProp; } });
var boolean_1 = require("./prop-types/boolean");
Object.defineProperty(exports, "booleanProp", { enumerable: true, get: function () { return boolean_1.booleanProp; } });
var number_1 = require("./prop-types/number");
Object.defineProperty(exports, "numberProp", { enumerable: true, get: function () { return number_1.numberProp; } });
var integer_1 = require("./prop-types/integer");
Object.defineProperty(exports, "integerProp", { enumerable: true, get: function () { return integer_1.integerProp; } });
var symbol_1 = require("./prop-types/symbol");
Object.defineProperty(exports, "symbolProp", { enumerable: true, get: function () { return symbol_1.symbolProp; } });
var vueComponent_1 = require("./prop-types/vueComponent");
Object.defineProperty(exports, "vueComponentProp", { enumerable: true, get: function () { return vueComponent_1.vueComponentProp; } });
var any_1 = require("./prop-types/any");
Object.defineProperty(exports, "anyProp", { enumerable: true, get: function () { return any_1.anyProp; } });
var array_1 = require("./prop-types/array");
Object.defineProperty(exports, "arrayProp", { enumerable: true, get: function () { return array_1.arrayProp; } });
var object_1 = require("./prop-types/object");
Object.defineProperty(exports, "objectProp", { enumerable: true, get: function () { return object_1.objectProp; } });
var function_1 = require("./prop-types/function");
Object.defineProperty(exports, "functionProp", { enumerable: true, get: function () { return function_1.functionProp; } });
var oneOf_1 = require("./prop-types/oneOf");
Object.defineProperty(exports, "oneOfProp", { enumerable: true, get: function () { return oneOf_1.oneOfProp; } });
var oneOfObjectKeys_1 = require("./prop-types/oneOfObjectKeys");
Object.defineProperty(exports, "oneOfObjectKeysProp", { enumerable: true, get: function () { return oneOfObjectKeys_1.oneOfObjectKeysProp; } });
var oneOfTypes_1 = require("./prop-types/oneOfTypes");
Object.defineProperty(exports, "oneOfTypesProp", { enumerable: true, get: function () { return oneOfTypes_1.oneOfTypesProp; } });
var instanceOf_1 = require("./prop-types/instanceOf");
Object.defineProperty(exports, "instanceOfProp", { enumerable: true, get: function () { return instanceOf_1.instanceOfProp; } });
var isNegative_1 = require("./validators/isNegative");
Object.defineProperty(exports, "isNegative", { enumerable: true, get: function () { return isNegative_1.isNegative; } });
var isPositive_1 = require("./validators/isPositive");
Object.defineProperty(exports, "isPositive", { enumerable: true, get: function () { return isPositive_1.isPositive; } });
var isNonNegative_1 = require("./validators/isNonNegative");
Object.defineProperty(exports, "isNonNegative", { enumerable: true, get: function () { return isNonNegative_1.isNonNegative; } });
var isNonPositive_1 = require("./validators/isNonPositive");
Object.defineProperty(exports, "isNonPositive", { enumerable: true, get: function () { return isNonPositive_1.isNonPositive; } });
